esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -1881,49 +1881,159 @@ IPropertyTree *CFileSprayEx::getAndValidateDropZone(const char *path, const char
     return nullptr;
 }
 
-void CFileSprayEx::readAndCheckSpraySourceReq(MemoryBuffer& srcxml, const char* srcIP, const char* srcPath, const char* srcPlane,
-    StringBuffer& sourcePlaneReq, StringBuffer& sourceIPReq, StringBuffer& sourcePathReq)
+static void checkHostInSpraySourcePath(const char* fileReq, const char* dropZoneName, IPropertyTree* dropZone, const char* sourceHostReq, SocketEndpoint* sourceHostEp, StringBuffer& localPath, StringBuffer& hostInPathStr)
 {
-    StringBuffer sourcePath(srcPath);
-    sourcePath.trim();
-    if(srcxml.length() == 0)
-    {
-        if (sourcePath.isEmpty())
-            throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "Source path not specified.");
-        if (containsRelPaths(sourcePath)) //Detect a path like: a/../../../f
-            throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "Invalid path %s", sourcePath.str());
-
-        if (!isEmptyString(srcPlane))
-        {
-            Owned<IPropertyTree> dropZone = getDropZonePlane(srcPlane);
-            if (!dropZone)
-                throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "Unknown landing zone: %s", srcPlane);
-            const char * dropZonePlanePath = dropZone->queryProp("@prefix");
-            if (isAbsolutePath(sourcePath))
-            {
-                if (!startsWith(sourcePath, dropZonePlanePath))
-                    throw makeStringException(ECLWATCH_INVALID_INPUT, "Invalid source path");
-            }
-            else
-            {
-                StringBuffer s(sourcePath);
-                sourcePath.set(dropZonePlanePath);
-                addNonEmptyPathSepChar(sourcePath);
-                sourcePath.append(s);
-            }
-            getDropZoneHost(srcPlane, dropZone, sourceIPReq);
-            sourcePlaneReq.append(srcPlane);
+    if (!isPathSepChar(fileReq[0]) | (fileReq[0] != fileReq[1]))
+        return;
+
+    splitUNCFilename(fileReq, &hostInPathStr, &localPath, &localPath, &localPath);
+    const char* hostInPath = hostInPathStr.str() + 2;
```

Comment:
needs/using bit after '//', i.e. the <hostname>, but returning (in hostInPathStr) the //<hostname> ..
that is confusing.
Should return the hostname - that if anything will be whats needed, caller should have to do ... + 2

Code clarity here is far more important than a slight gain in efficiency because need to copy in to output StringBuffer.